### PR TITLE
Add ability to set custom initiating flows for NotaryService

### PR DESCRIPTION
**Pull request by alexeykoren**
Currently notary service developer cannot specify custom client initiating flow. NotaryService.Client is hardcoded in AbstractNode.

New way is proposed in this PR that allows:

1) Define custom client flow
2) Define more than 1 client flow that can be served by Notary. This might be needed, for instance, if Notary is serving both "normal" client and clients that use custom flows (e.g. privacy-preserving verification).

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryService.kt [hunk 1]
```diff
@@ -14,6 +14,14 @@ abstract class NotaryService : SingletonSerializeAsToken() {
     abstract val services: ServiceHub
     abstract val notaryIdentityKey: PublicKey
 
+    /**
+     * Maps initiating flow classes to factory methods than create responder flows.
```
Comment (blsemo) [inline core/src/main/kotlin/net/corda/core/internal/notary/NotaryService.kt]: I am sure there is a typo in that comment - I can't quite parse it.

**Review by blsemo — COMMENTED** [review]
Looks fine in principle, but could do with a better comment.

**Comment by relyafi** [comment]
@alexeykoren I believe @joeldudleyr3 has already spoken with you regarding release scheduling for this change. Please can you rebase the changes against `release/os/4.8`?

File: core/src/main/kotlin/net/corda/core/internal/notary/NotaryService.kt (hunk 1 above)
Comment (alexeykoren) [reply core/src/main/kotlin/net/corda/core/internal/notary/NotaryService.kt]: Sure, that was definitely a typo, thanks for spotting! I've updated the comment. Is it more clear now?

**Comment by alexeykoren** [comment]
> @alexeykoren I believe @joeldudleyr3 has already spoken with you regarding release scheduling for this change. Please can you rebase the changes against `release/os/4.8`?

@relyafi I've rebased to 4.8 but it doesn't look really clear. I guess now its easier to recreate the merge request based on 4.8. What do you think?

**Comment by relyafi** [comment]
> > @alexeykoren I believe @joeldudleyr3 has already spoken with you regarding release scheduling for this change. Please can you rebase the changes against `release/os/4.8`?
> 
> @relyafi I've rebased to 4.8 but it doesn't look really clear. I guess now its easier to recreate the merge request based on 4.8. What do you think?

@alexeykoren are you not able to rebase against 4.8 such that your change is the latest change? It also looks like your branch now have conflicts that need to be resolved which may go away after doing this.

If that doesn't work, then yes a new merge request would be another option.

**Comment by StefanIliev545** [comment]
@alexeykoren the change looks good. Before being merged it needs to have support for a config field to enable/disable it. Reference PR I hacked: https://github.com/corda/corda/pull/6843

**Comment by alexeykoren** [comment]
@relyafi I've cleaned the branch up, rebased on top of 4.8 and also added @StefanIliev545 's suggestions regarding configuration

**Review by StefanIliev545 — DISMISSED** [review]
LGTM. 
PR needs a fix for the nesting issue detekt is complaining about, otherwise tests are good.
[2021-01-16T12:08:08.113Z] 	NestedBlockDepth - 4/4 - [maybeStartNotaryService] at /var/jenkins/workspace/equest-Code-Checks_corda_PR-6832/node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt:1021:5

**Comment by relyafi** [comment]
@alexeykoren thanks for addressing the comments. Please can you address the detekt issue that has been raised, then @StefanIliev545 can re-trigger the checks and we should be able to get this in.

**Review by StefanIliev545 — APPROVED** [review]
LGTM, tests passed.

**Review by blsemo — APPROVED** [review]

**Pull request merged**